virtcontainers/network.go

```diff
@@ -443,6 +462,138 @@ func (endpoint *PhysicalEndpoint) HotDetach(h hypervisor, netNsCreated bool, net
 	return fmt.Errorf("PhysicalEndpoint does not support Hot detach")
 }
 
+// Macvlan
```

Comment:
`network.go` is huge - without this PR, it's the largest non-`_test.go` file. With this PR, it is the largest `*.go` file.

I guess it shouldn't block landing the PR, but I'd strongly recommend we split this file into manageable chunks:

- `network.go`
- `network_physical.go`
- `network_vhost.go`
- *etc*

Reply:
+1 for this proposal. But I would recommend something slightly different. The `network.go` file declare both the `network` interface and the `endpoint` interface.
We should have:
- `network.go`, declaring `network` interface and any generic function related to network.
- `default_network.go` as an implementation of the `network` interface. (here already)
- `noop_network.go` as another implementation of the `network` interface. (here already)
- `endpoint.go` declaring `endpoint` interface and generic functions related to endpoints.
- `macvlan_endpoint.go` defining the MACVLAN implementation.
- `vfio_endpoint.go` defining the VFIO implementation.
and so on...

Reply:
Thanks @sboeuf - that indeed looks better! :)

Reply:
@jodh-intel @sboeuf Yup, I realized we need to refactor a lot while working on this PR. I intend to do all of that in a follow up PR. That includes moving each endpoint in its own go file. I did not want to make this PR any larger. I am going to create an issue for the refactor.

Reply:
@jodh-intel @sboeuf https://github.com/kata-containers/runtime/issues/799

Reply:
Thanks @amshinde! :)